ind.ie.Gnomit.json

```diff
@@ -0,0 +1,109 @@
+{
+    "app-id" : "ind.ie.Gnomit",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.28",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "ind.ie.Gnomit",
+    "x-run-args" : [
+        "sandbox/gnomit/gjs/tests/message-with-body"
+    ],
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--filesystem=host",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options" : {
+        "cflags" : "-O2 -g",
+        "cxxflags" : "-O2 -g",
+        "env" : {
+            "V" : "1"
+        }
+    },
+    "cleanup" : [
+        "/include",
+        "/lib/pkgconfig",
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+            "name" : "gspell",
+            "config-opts" : [
+                "--disable-vala",
+                "--disable-static",
+                "--disable-gtk-doc"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/gspell.git",
+                    "tag" : "1.6.1",
+                    "commit" : "f2823c994b9af57d91e75c54bd1ec68a1dee5767"
+                }
+            ],
+            "cleanup" : [
+                "/bin"
+            ]
+        },
+        {
+            "name" : "icons",
```

Comment:
Hi Ned, do you mean the icons? (Assuming your comments are focussed on the last line of the listing). This is the only way (after hours of searching for any documentation) that I managed to get the icons to show up when running the app via Flatpak. I’m afraid I don’t know right now how to add them as part of the post-install step or what “properly from within your app” means. Do you have any links to relevant documentation handy? I will set aside a few more hours to try and find some documentation and fix this. Once the app is acceptable/accepted, I plan to write up tutorials on all this as “getting started” documentation is severely lacking at the moment.

Reply:
OK, I think I found something I can try – how Gnome Recipes is doing it: https://salsa.debian.org/gnome-team/gnome-recipes/blob/debian/master/data/meson.build

Reply:
Sure, more documentation is always welcome! A lot of flatpak stuff assumes you have "a linux app" already, and is just silent about where you're actually supposed to put stuff. I'm delighted that this is getting more people to write new things, but obviously embarassed it's not as easy as it should be.

What I meant is - you could include these assets in your app and use meson to put them in the correct place. Alternatively, rather than having them as a separate module in flatpak, include them as part of the main app module and use a "post-install" section (which is basically like buildsystem simple, only it runs after the main meson build but before you move onto the next section. Here's an example of someone doing exactly that:

https://github.com/flathub/space.fips.Fips/blob/30038cb77d9f258b5bc0f044b0986ab93f6f7cdb/space.fips.Fips.json#L28

Reply:
Thanks, Nick – I got working now. Should be able to issue a PR to address all these issues within half an hour or so :)

My goal is to make Gnomit a little case study on how to take an app from zero to Flatpak/Flathub. Should be just about possible given how simple it is. I plan on having a GJS version and then doing one for Vala when the Vala version is ready. (These are learning exercises for me also.)